### PR TITLE
F sharp templates

### PR DESCRIPTION
**Pull request by jamessdixon**
swtiched out some if.. then for pattern matches.  Changed 2 loops to a Seq.iter

### CODE REVIEW

**Comment by soninaren** [comment]
@sylvanc can you take a look?

File: Templates.sln [hunk 1]
```diff
@@ -656,7 +656,7 @@ Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "TimerTrigger-FSharp", "Time
 EndProject
 Global
 	GlobalSection(SolutionProperties) = preSolution
-		HideSolutionJavaScript = FALSE
```
Comment (christopheranderson) [inline Templates.sln]: Why was this changed?

Comment (jamessdixon) [inline Templates.sln]: Magic of Visual Studio?  That was not me (though hiding javascript is usually not a bad idea).  Do you want me to fix and resubmit?

**Comment by sylvanc** [comment]
LGTM :)

**Comment by dsyme** [comment]
These changes are not necessary, though thanks for proposing them

* while some F# programmers like to pattern match on single boolean true/false values, i'ts not particularly idiomatic to do that - indeed I suppose if/then/else would normally be seen as marginally more idiomatic, 

* For Seq.iter, much the same applies: some F# programmers eschew the use of "for". However there can also be good reasons to use it, notably the way it combines with computation expressions and comprehensions.

cheers and thanks
don

**Comment by jamessdixon** [comment]
Who is this Don Syme and what does he know about F#?  Granted the changes aren't necessary, I have found that syntax like that does help when transitioning people from imperative constructs into a more functional style.  Since these templates are going into Visual Studio and a majority of VS Users are C# devs, it seemed like a easy point of education.

**Comment by dsyme** [comment]
@jamessdixon :) I understand - yes, it's tricky balance between familiarity (more like C#) and teaching (use more F#-specific constructs). 

However for these templates I think the emphasis shouldn't be on teaching F#, just on landing people with an Azure Function that does the job for them

**Comment by jamessdixon** [comment]
I was going to add some more templates.  Can I use that style or do you want consistency in the code base?

**Comment by dsyme** [comment]
@jamessdixon Consistency please (unless that prevents you adding the templates :) :) :) )

**Comment by jamessdixon** [comment]
The empty template in Visual Studio right now has an unneeded semi colon:

let Run(log: TraceWriter ) =
    log.Verbose("F# function executed.")**;**

Want a pull request for that?

**Comment by soninaren** [comment]
Visual studio is using older version F# templates, All empty templates have been removed from the repo. Also FSharpTemplates branch is no longer in use, It was created to temporarily hold some F# Templates. Please use send all future PR's against dev branch. :)

**Comment by jamessdixon** [comment]
OK - will do
When will these templates be pushing into VS?  Should I ask the VS team?

**Comment by soninaren** [comment]
VS team controls that timeline, I will ask them and let you know.

**Comment by jamessdixon** [comment]
Thanks- I can also ask on the MVP thread if you don't get a fast answer

**Comment by christopheranderson** [comment]
Summoning @lindydonna who has the answer on VS stuff. 

FYI - our cli uses these templates when you run locally and it pulls from GitHub directly.

**Comment by lindydonna** [comment]
@jamessdixon We don't have a concrete timeline on when there will be a refresh of the Azure Functions Visual Studio tools. Currently it uses a snapshot of the templates feed from 0.6. We're changing up how the tooling works, which is why I don't have anything more concrete. :)

**Comment by soninaren** [comment]
From the discussion above it looks like we might not be merging the changes, would it be okay if we close this PR?

**Comment by jamessdixon** [comment]
Sure - though I think Don is making a mistake.  Let the records show that I shook my fist at him.  Thanks!

**Comment by soninaren** [comment]
:)
